op_crates/fetch/lib.rs

```diff
@@ -409,30 +419,48 @@ where
   }
 
   let client =
-    create_http_client(args.ca_file.as_deref(), args.ca_data.as_deref())
+    create_http_client2(args.ca_file.as_deref(), args.ca_data.as_deref())
       .unwrap();
 
   let rid = state.resource_table.add(HttpClientResource::new(client));
   Ok(json!(rid))
 }
 
-/// Create new instance of async reqwest::Client. This client supports
-/// proxies and doesn't follow redirects.
-fn create_http_client(
+// Internal version used by ops
+fn create_http_client2(
   ca_file: Option<&str>,
   ca_data: Option<&str>,
 ) -> Result<Client, AnyError> {
-  let mut builder = Client::builder().redirect(Policy::none()).use_rustls_tls();
-  if let Some(ca_data) = ca_data {
-    let ca_data_vec = ca_data.as_bytes().to_vec();
-    let cert = reqwest::Certificate::from_pem(&ca_data_vec)?;
-    builder = builder.add_root_certificate(cert);
+  let ca_data_vec = if let Some(ca_data) = ca_data {
+    Some(ca_data.as_bytes().to_vec())
   } else if let Some(ca_file) = ca_file {
     let mut buf = Vec::new();
     File::open(ca_file)?.read_to_end(&mut buf)?;
-    let cert = reqwest::Certificate::from_pem(&buf)?;
+    Some(buf)
+  } else {
+    None
+  };
+  create_http_client("".to_owned(), ca_data_vec)
```

Comment:
This sets `User-Agent` to an empty string, which doesn't seem right. Why split into `create_http_client` two functions?

Reply:
@bartlomieju There are already two duplicate`create_http_client` functions in `main`:
1. [runtime/http_util.rs](https://github.com/denoland/deno/blob/20627c91364d2a992fdfaaad7c8ae86454dbc2ed/runtime/http_util.rs#L13)
2. [op_crates/fetch/lib.rs](https://github.com/denoland/deno/blob/20627c91364d2a992fdfaaad7c8ae86454dbc2ed/op_crates/fetch/lib.rs#L412)

Both are broadly used for the fetch ops, the first is put into OpState the latter is used in `op_create_http_client`. So even in `main`, `op_create_http_client` doesn't allow setting the user-agent nor does it appear to use the "global" one from the OpState. `op_create_http_client` doesn't appear to be used anywhere so I was considering stripping it out or fixing it so it shares the user-agent.

Reply:
@AaronO `op_create_http_client` is used for `Deno.createHttpClient`.

Reply:
@lucacasonato Yep, sorry if I was unclear. I meant that I didn't see any calls to `Deno.createHttpClient` in our code or elsewhere (only [results on sourcegraph](https://sourcegraph.com/search?q=Deno.createHttpClient&patternType=literal) are deno's own tests).

But I would still argue that it's "incorrect" and should either accept a `userAgent` arg or default to the one provide to the default client.

Reply:
It is indeed incorrect - the default user agent should be the same as the default http client. If it is not, that is a bug. Users can already specify a custom user agent by specifying the `User-Agent` header on requests.

Reply:
I created an issue and provided a reproducible example here: https://github.com/denoland/deno/issues/9827, so it is confirmed, I'll fix it.

I do think being able to set `User-Agent` for a client would be convenient, though it can be worked around as you said, it's usually the kind of thing that is provided by the client/user-agent.

Reply:
Fixed by https://github.com/denoland/deno/pull/9830